### PR TITLE
serial: sam0: Conversion of k_work API

### LINKED ISSUE #34102: Convert drivers/serial/uart_sam0.c to new kwork API
Convert drivers/serial/uart_sam0.c to use new kwork API.

See the following issue for details:

#33104

### PR DESCRIPTION
**Pull request by nandojve**
Replace all existing deprecated API with the recommended alternative.

Fixes #34102

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>
Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by mnkp — APPROVED** [review]

**Comment by carlescufi** [comment]
Waiting for a review from @pabigot

**Review by pabigot — APPROVED** [review]

**Pull request merged**
